### PR TITLE
documents 08_vision.data: TransformBlocks for vision

### PR DESCRIPTION
**Pull request by ChrisRichardMiles**
The only issue is that I changed an argument from `cls` to `clss`.  The problem is that show_doc does not show the "cls" argument. You can see that in the _format_args function [here]https://github.com/fastai/nbdev/blob/master/nbdev/showdoc.py#L347. I know we aren't supposed to touch any code, but I'm not sure what to do here.

### CODE REVIEW

**Review by warner-benjamin — CHANGES_REQUESTED** [review]
> The only issue is that I changed an argument from `cls` to `clss`

Yeah, this is a nonstarter. As you can see from the CI output, it broke existing code. If there's a bug in nbdev, it'll need to be fixed there.

Try installing the latest nbdev with bugfixes from github
```
pip install git+https://github.com/fastai/nbdev -U
```
and see if that resolves the issue. Otherwise, mention it and this PR in the sprint-chat.

**Comment by muellerzr** [comment]
@ChrisRichardMiles please change this back to `cls`. We'll be addressing the cls issue in nbdev

File: fastai/vision/data.py [hunk 1]
```diff
@@ -94,7 +98,10 @@ def MaskBlock(codes=None):
 BBoxBlock.__doc__  = "A `TransformBlock` for bounding boxes in an image"
 
 # Cell
-def BBoxLblBlock(vocab=None, add_na=True):
+def BBoxLblBlock(
+    vocab=None, # List of possible labels for bounding boxes
+    add_na:bool=True # If `True`, a new category is added for NaN (that will represent the background class)
+):
```
Comment (warner-benjamin) [inline fastai/vision/data.py:104]:
```suggestion
def BBoxLblBlock(
    vocab:list=None, # Vocab labels for bounding boxes
    add_na:bool=True # Add NaN as a background class
):
```

File: fastai/vision/data.py [hunk 2]
```diff
@@ -77,12 +77,16 @@ def show_batch(x:TensorImage, y:TensorImage, samples, ctxs=None, max_n=10, nrows
...
+    cls=PILImage # A class that inherits from `PILBase`
+):
     "A `TransformBlock` for images of `cls`"
     return TransformBlock(type_tfms=cls.create, batch_tfms=IntToFloatTensor)
 
 # Cell
-def MaskBlock(codes=None):
+def MaskBlock(
+    codes=None # List of classes where `codes[i]` corresponds to an `i` valued pixel in a segmentation mask
+):
```
Comment (warner-benjamin) [inline fastai/vision/data.py:89]:
```suggestion
def MaskBlock(
    codes:list=None # Vocab labels for segmentation masks
):
```

**Review by warner-benjamin — CHANGES_REQUESTED** [review]
Please add back the deleted doc cell. Also, don't check changes to other notebooks then the one you are working on.

Some suggestions for doc string improvements, replicate these to the notebook and resolve the issues mentioned above and it will be ready to merge.

File: fastai/vision/data.py [hunk 3]
```diff
@@ -77,12 +77,16 @@ def show_batch(x:TensorImage, y:TensorImage, samples, ctxs=None, max_n=10, nrows
     return ctxs
 
 # Cell
-def ImageBlock(cls=PILImage):
+def ImageBlock(
+    cls=PILImage # A class that inherits from `PILBase`
+):
```
Comment (marii-moe) [inline fastai/vision/data.py:82]:
```suggestion
def ImageBlock(cls:PILBase=PILImage):
```
Can just add that as a typehint instead.

**Review by marii-moe — CHANGES_REQUESTED** [review]

**Comment by warner-benjamin** [comment]
@ChrisRichardMiles the changes need to be added to the `08_vision.data.ipynb` notebook and the unchanged notebooks reverted.

**Comment by ChrisRichardMiles** [comment]
> @ChrisRichardMiles the changes need to be added to the `08_vision.data.ipynb` notebook and the unchanged notebooks reverted.

Ok I will do it first thing in the morning. Thanks for the help

**Comment by ChrisRichardMiles** [comment]
@warner-benjamin @marii-moe 
I have made the suggested changes and reverted the notebooks that should not have been touched. I still have an annoying local error message that I haven't been able to fix yet. I will fix it as soon as I can (hopefully tonight) and then I will request reviews.

**Review by warner-benjamin — APPROVED** [review]
@ChrisRichardMiles Assuming the"UserWarning: __floordiv__ is deprecated" is what you are referring to, it was already checked into the codebase, so you don't need to worry about it. There is an issue open for it #3588.

LGTM. Thanks!

**Comment by ChrisRichardMiles** [comment]
> @ChrisRichardMiles Assuming the"UserWarning: __floordiv__ is deprecated" is what you are referring to, it was already checked into the codebase, so you don't need to worry about it. There is an issue open for it #3588.
> 
> 
> 
> LGTM. Thanks!

No, I actually meant an error that I also had in my last PR. But looking again, I realize I misread the file changes, because that part is highlighted in red, so it seems it was deleted. I am still confused to see it at all, but I guess everything is ok, so I'm going to request the review of @marii-moe

**Review by marii-moe — APPROVED** [review]
LGTM

**Pull request merged**